### ISSUE TITLE
RANS simulations on 3D wing - convergence problem

### ISSUE THREAD

**Issue by acrovato** [su2code/SU2#533]
Dear SU2 developers and users,

I am running RANS simulations around a typical aircraft wing (EBW in the following) and I am facing some issues.

Here are the flight conditions:
- Mach number: 0.78
- Temperature = ~217 K
- Reynolds number: ~19 millions
- AoA = 0°

At these conditions, the EBW should produce a lift coefficient between [0.4;0.5].
However, when SU2 converges, the Cl is about 0.27 and the Cd is negative.

Here follows a summary of what I have done.

I have used several (multiblock structured) meshes:
- Gmsh (1M cells), SU2 and CGNS format
- ICEM (10M cells), CGNS format

At the preprocessing stage when computing the projection area of the EBW in the z-plane, I noticed that the meshes generated by Gmsh gave a wrong projection (~30 m2 instead of ~40 m2!).
I increased the height of the first cell in the boundary layer (which was set to match y+ ~ 1) and the problem disappeared when y+ got close to 30. Of course, this mesh is not usable without wall functions...
However, it does not seem to be the only problem since:
- the ICEM mesh does not converge to the expected value of Cl
- a colleague ran both (Gmsh and ICEM) meshes with identical settings on the commercial solver CFD++ and got the expected results (Cl = [0.4;0.5]).

I also tried to tune these different numerical parameters, without success:
- ROE (2nd order with Vankatakrishnan/Vankatakrishnan-Wang limiter) and JST numerical schemes
- Green-Gauss or Weighted Least Square for gradients
- LU_SGS and ILU linear solver preconditionners

To make sure that SU2 was correctly compiled, I also:
- ran those simulations on SU2 v6.0.0 and v4.3.0, both compiled on Linux. I got the same results in both cases
- ran the Onera M6 test case provided with SU2
- created the same mesh that I used for the EBW with my Gmsh script but around the Onera M6 (I just changed the height of the first cell which is smaller for the M6 than for my wing). I recovered the expected results

Finally, I also ran the case with the Euler equations, on the EBW meshed with Gmsh, and recovered the expected results.

In summary, I have tried several meshes from different software, with different numerical parameters on different SU2 versions, but always got bad results when solving RANS equations.
Particularly, the Gmsh mesh preprocessing on SU2 gave a bad z-projection of the EBW.
However, the same meshes gave converged results on CFD++.

Any help or ideas would be appreciated.
Many thanks in advance.

**Comment by vdweide** [comment]
Would it be possible to make available the Gmsh mesh and the config file?

**Comment by hlkline** [comment]
I noticed you mentioned that it is a multiblock structured mesh. SU2 expects a single block format, unless explicitly setting up a  multi zone problem, so there's an outside chance it's only loading part of the mesh. I would expect that to throw some kind of error, but it's worth checking.
This could explain the wrong z projection, but to check you can plot the output volume solution to see if your mesh is loaded appropriately.

**Comment by AdrienCro** [comment]
@vdweide: I cannot give you the geometry or the mesh because they are the property of an aircraft manufacturer, a colleague is going to check if he can provide a dummy geometry, similar to the one we are working onto. What I can give you is the config file (I still had to change some values related to the geometry).

@hlkline: What I meant was that the meshes were generated using a multi block strategy. Before writing into SU2 or CGNS, they were converted in a proper unstructured format. I had indeed checked the surface solution, but not the volume one. And it seems that there are some weird cells (see attached screenshot). I will investigate.

Thanks for your inputs!

[EBW2v2.txt](https://github.com/su2code/SU2/files/1999732/EBW2v2.txt)

![volume](https://user-images.githubusercontent.com/39187559/39983910-a25a9172-5759-11e8-9d5e-d7066ae95d00.png)

**Comment by vdweide** [comment]
The config file is a start, but for a thorough investigation a full test case, i.e. including a grid, is preferable.

**Comment by acrovato** [comment]
Dear SU2 users,

I continued the investigation and managed to track down the source of the problem.
The issues I face with SU2 appear when my wing geometry has several airfoils across the span.

I ran several cases with supercritical airfoils and NACA 4 digits airfoils. As soon as the airfoil was changing along the span, SU2 had troubles  computing the projected and wetted area of the wing, an attaining a correct solution.

To better illustrate the issue, I attached a SU2 .cfg file (dummy.txt) and Gmsh .geo file (dummy_mesh.txt) to this post.
The geometry is the same as in the previous post except that I used 3 different airfoils along the span. That is, I replaced the NASA SC(2)-0712 by the {NASA SC(2)-0714, NASA SC(2)-0712, NASA SC(3)-0712}.
These airfoils are similar, so I expected to recover similar results. However, I observed that:
- SU2 computed the wrong z-projected area : 23.1 m^2 instead of 38 m^2,
- the y+ of the first cell is way too high in certain regions (see attached image, grey zones are the area where y+<1),
- the Cp distribution if not looking good (makes sense with those values of y+), see attached file.

I am unsure if the problem appears because Gmsh has trouble in generating this kind of mesh or because SU2 has trouble reading it.

Any ideas are welcome.

Thank you again,
Adrien

![cp](https://user-images.githubusercontent.com/39187559/43064280-c076a5e2-8e5e-11e8-84e5-602d1ee27ea2.png)
![y](https://user-images.githubusercontent.com/39187559/43064281-c09a1f86-8e5e-11e8-82b9-4c4fddc7d470.png)
[dummy_mesh.txt](https://github.com/su2code/SU2/files/2218394/dummy_mesh.txt)
[dummy.txt](https://github.com/su2code/SU2/files/2218395/dummy.txt)
[log.txt](https://github.com/su2code/SU2/files/2218411/log.txt)
